test/e2e/sync_options_test.go

```diff
@@ -173,6 +175,34 @@ func TestSyncWithApplyOutOfSyncOnly(t *testing.T) {
 		Expect(ResourceResultIs(ResourceResult{Group: "apps", Version: "v1", Kind: "Deployment", Namespace: ns, Name: "guestbook-ui", Message: "deployment.apps/guestbook-ui configured", SyncPhase: SyncPhaseSync, HookPhase: OperationRunning, Status: ResultCodeSynced, Images: []string{"quay.io/argoprojlabs/argocd-e2e-container:0.2"}}))
 }
 
+func TestCliSyncOverridesApplicationSpecSyncApplyOutOfSyncOnlyOptions(t *testing.T) {
+	// This test covers the following case: App spec has ApplyOutOfSyncOnly=true, but CLI sync without sync options is used.
+	// In this case, all resources should be synced as CLI sync options take precedence over the Application spec sync options.
```

Comment:
This behavior is a little surprising to me... I'd expect "unspecified" to mean "default to what's set on the App." Would it be possible to enable that behavior with nilable flags? Basically if any sync option is set (a non-nil bool), then we construct a sync options object. Otherwise we send nil and let the server decide what to do.

Reply:
Well currently it is like that, "unspecified" means "default to what's set on the App" - but this is the root cause of the issue that you cannot override the App spec with empty options from CLI.
To me this is counter-intuitive, since in the UI I can uncheck all the boxes (that represent the App spec) and then have the sync done with empty options, so I'd expect the CLI to behave the same... I guess the question is similar to  comparing `patch` and `update` with empty settings :)
And so I changed this behavior in the PR to override with empty settings.
Could you elaborate more on the nilable flags?

Reply:
A kinda funny option would be to recommend folks use `--server-side-apply`, which would cause all the sync options to be overridden. :-)